### DISCUSSION TITLE
how to add udp noise to warp config?

### DISCUSSION THREAD

**Discussion by developer861** [XTLS/Xray-core#3974] [Q&A] [answered]
i want to add udp noise freedom to this warp config can you help?

{
        "remarks": "💦 29 - WoW 🌍",
        "log": {
            "loglevel": "warning"
        },
        "dns": {
            "hosts": {
                "domain:googleapis.cn": [
                    "googleapis.com"
                ],
                "geosite:category-ads-all": [
                    "127.0.0.1"
                ],
                "geosite:category-ads-ir": [
                    "127.0.0.1"
                ]
            },
            "servers": [
                "1.1.1.1",
                "1.0.0.1",
                {
                    "address": "94.140.14.14",
                    "domains": [
                        "geosite:category-ir"
                    ],
                    "expectIPs": [
                        "geoip:ir"
                    ]
                }
            ],
            "tag": "dns"
        },
        "inbounds": [
            {
                "port": 10808,
                "protocol": "socks",
                "settings": {
                    "auth": "noauth",
                    "udp": true,
                    "userLevel": 8
                },
                "sniffing": {
                    "destOverride": [
                        "http",
                        "tls"
                    ],
                    "enabled": true,
                    "routeOnly": true
                },
                "tag": "socks-in"
            },
            {
                "port": 10809,
                "protocol": "http",
                "settings": {
                    "auth": "noauth",
                    "udp": true,
                    "userLevel": 8
                },
                "sniffing": {
                    "destOverride": [
                        "http",
                        "tls"
                    ],
                    "enabled": true,
                    "routeOnly": true
                },
                "tag": "http-in"
            },
            {
                "listen": "127.0.0.1",
                "port": 10853,
                "protocol": "dokodemo-door",
                "settings": {
                    "address": "1.1.1.1",
                    "network": "tcp,udp",
                    "port": 53
                },
                "tag": "dns-in"
            }
        ],
        "outbounds": [
            {
                "protocol": "wireguard",
                "settings": {
                    "address": [
                        "172.16.0.2/32",
                        "2606:4700:110:8a88:2791:a8d0:e091:c395/128"
                    ],
                    "mtu": 1280,
                    "peers": [
                        {
                            "endpoint": "162.159.192.76:1843",
                            "publicKey": "",
                            "keepAlive": 5
                        }
                    ],
                    "reserved": [
                        190,
                        2,
                        236
                    ],
                    "secretKey": ""
                },
                "streamSettings": {
                    "sockopt": {
                        "dialerProxy": "proxy",
                        "tcpKeepAliveIdle": 100,
                        "tcpNoDelay": true
                    }
                },
                "tag": "chain"
            },
            {
                "protocol": "wireguard",
                "settings": {
                    "address": [
                        "172.16.0.2/32",
                        "2606:4700:110:8336:4d37:8de4:cfd2:90fb/128"
                    ],
                    "mtu": 1280,
                    "peers": [
                        {
                            "endpoint": "",
                            "publicKey": "",
                            "keepAlive": 5
                        }
                    ],
                    "reserved": [
                        241,
                        125,
                        57
                    ],
                    "secretKey": ""
                },
                "tag": "proxy"
            },
            {
                "protocol": "dns",
                "tag": "dns-out"
            },
            {
                "protocol": "freedom",
                "settings": {},
                "tag": "direct"
            },
            {
                "protocol": "blackhole",
                "settings": {
                    "response": {
                        "type": "http"
                    }
                },
                "tag": "block"
            }
        ],
        "policy": {
            "levels": {
                "8": {
                    "connIdle": 300,
                    "downlinkOnly": 1,
                    "handshake": 4,
                    "uplinkOnly": 1
                }
            },
            "system": {
                "statsOutboundUplink": true,
                "statsOutboundDownlink": true
            }
        },
        "routing": {
            "domainStrategy": "IPIfNonMatch",
            "rules": [
                {
                    "inboundTag": [
                        "dns-in"
                    ],
                    "outboundTag": "dns-out",
                    "type": "field"
                },
                {
                    "inboundTag": [
                        "socks-in",
                        "http-in"
                    ],
                    "port": "53",
                    "outboundTag": "dns-out",
                    "type": "field"
                },
                {
                    "ip": [
                        "94.140.14.14"
                    ],
                    "port": "53",
                    "outboundTag": "direct",
                    "type": "field"
                },
                {
                    "domain": [
                        "geosite:private",
                        "geosite:category-ir"
                    ],
                    "outboundTag": "direct",
                    "type": "field"
                },
                {
                    "ip": [
                        "geoip:private",
                        "geoip:ir"
                    ],
                    "outboundTag": "direct",
                    "type": "field"
                },
                {
                    "network": "udp",
                    "port": "443",
                    "outboundTag": "block",
                    "type": "field"
                },
                {
                    "domain": [
                        "geosite:category-ads-all",
                        "geosite:category-ads-ir"
                    ],
                    "outboundTag": "block",
                    "type": "field"
                },
                {
                    "network": "tcp,udp",
                    "outboundTag": "chain",
                    "type": "field"
                }
            ]
        },
        "stats": {}
    }

**Comment by dragonbreath2000** [comment, marked as answer]
https://github.com/bia-pain-bache/BPB-Worker-Panel/issues/463#issuecomment-2462944854

**Comment by diarrassoubaib62-design** [comment]
{"remarks":"","log":{"loglevel":"warning"},"inbounds":[{"tag":"socks","port":10808,"protocol":"socks","settings":{"auth":"noauth","udp":true,"userLevel":8},"sniffing":{"enabled":true,"destOverride":["http","tls"],"routeOnly":false}}],"outbounds":[{"tag":"proxy","protocol":"vmess","settings":{"vnext":[{"address":"com.com","port":443,"users":[{"id":"99528a03-838c-4993-82b6-75913199ae34","security":"auto","level":8}]}]},"streamSettings":{"network":"ws","security":"tls","wsSettings":{"path":"/DANOUA","headers":{"Host":"danouasv.donigroup.xyz"}},"tlsSettings":{"allowInsecure":true,"serverName":"danouasv.donigroup.xyz","show":false}},"mux":{"enabled":false,"concurrency":-1,"xudpConcurrency":8,"xudpProxyUDP443":""}},{"tag":"direct","protocol":"freedom","settings":{"domainStrategy":"UseIP"},"mux":{"enabled":false,"concurrency":8,"xudpConcurrency":8,"xudpProxyUDP443":""}},{"tag":"block","protocol":"blackhole","settings":{"response":{"type":"http"}},"mux":{"enabled":false,"concurrency":8,"xudpConcurrency":8,"xudpProxyUDP443":""}}],"dns":{"servers":["1.1.1.1"],"hosts":{"domain:googleapis.cn":"googleapis.com","dns.alidns.com":["223.5.5.5","223.6.6.6","2400:3200::1","2400:3200:baba::1"],"one.one.one.one":["1.1.1.1","1.0.0.1","2606:4700:4700::1111","2606:4700:4700::1001"],"dot.pub":["1.12.12.12","120.53.53.53"],"dns.google":["8.8.8.8","8.8.4.4","2001:4860:4860::8888","2001:4860:4860::8844"],"dns.quad9.net":["9.9.9.9","149.112.112.112","2620:fe::fe","2620:fe::9"],"common.dot.dns.yandex.net":["77.88.8.8","77.88.8.1","2a02:6b8::feed:0ff","2a02:6b8:0:1::feed:0ff"]}},"routing":{"domainStrategy":"IPIfNonMatch","rules":[{"type":"field","ip":["1.1.1.1"],"outboundTag":"proxy","port":"53"},{"type":"field","ip":["223.5.5.5"],"outboundTag":"direct","port":"53"}]}}

**Reply by diarrassoubaib62-design** [reply]
{"remarks":"","log":{"loglevel":"warning"},"inbounds":[{"tag":"socks","port":10808,"protocol":"socks","settings":{"auth":"noauth","udp":true,"userLevel":8},"sniffing":{"enabled":true,"destOverride":["http","tls"],"routeOnly":false}}],"outbounds":[{"tag":"proxy","protocol":"vmess","settings":{"vnext":[{"address":"com.com","port":443,"users":[{"id":"99528a03-838c-4993-82b6-75913199ae34","security":"auto","level":8}]}]},"streamSettings":{"network":"ws","security":"tls","wsSettings":{"path":"/DANOUA","headers":{"Host":"danouasv.donigroup.xyz"}},"tlsSettings":{"allowInsecure":true,"serverName":"danouasv.donigroup.xyz","show":false}},"mux":{"enabled":false,"concurrency":-1,"xudpConcurrency":8,"xudpProxyUDP443":""}},{"tag":"direct","protocol":"freedom","settings":{"domainStrategy":"UseIP"},"mux":{"enabled":false,"concurrency":8,"xudpConcurrency":8,"xudpProxyUDP443":""}},{"tag":"block","protocol":"blackhole","settings":{"response":{"type":"http"}},"mux":{"enabled":false,"concurrency":8,"xudpConcurrency":8,"xudpProxyUDP443":""}}],"dns":{"servers":["1.1.1.1"],"hosts":{"domain:googleapis.cn":"googleapis.com","dns.alidns.com":["223.5.5.5","223.6.6.6","2400:3200::1","2400:3200:baba::1"],"one.one.one.one":["1.1.1.1","1.0.0.1","2606:4700:4700::1111","2606:4700:4700::1001"],"dot.pub":["1.12.12.12","120.53.53.53"],"dns.google":["8.8.8.8","8.8.4.4","2001:4860:4860::8888","2001:4860:4860::8844"],"dns.quad9.net":["9.9.9.9","149.112.112.112","2620:fe::fe","2620:fe::9"],"common.dot.dns.yandex.net":["77.88.8.8","77.88.8.1","2a02:6b8::feed:0ff","2a02:6b8:0:1::feed:0ff"]}},"routing":{"domainStrategy":"IPIfNonMatch","rules":[{"type":"field","ip":["1.1.1.1"],"outboundTag":"proxy","port":"53"},{"type":"field","ip":["223.5.5.5"],"outboundTag":"direct","port":"53"}]}}

**Comment by diarrassoubaib62-design** [comment]
![Uploading IMG_8758.png…]()
